File: FE/src/features/expense/services/expenseService.ts
import { apiClient } from '../../../services/apiClient'
import type { PaginatedResult } from '../../../types/api.types'
import type { CreateExpenseRequest, ExpenseResponse } from '../types/expense.types'

export const expenseService = {
  getAll: (page = 1, size = 10, sort = 'expenseDate,desc') => {
    const params = new URLSearchParams({
      page: String(page),
      size: String(size),
      sort,
    })
    return apiClient<PaginatedResult<ExpenseResponse>>(`/expenses?${params}`)
  },

  getById: (id: number) => apiClient<ExpenseResponse>(`/expenses/${id}`),

  create: (data: CreateExpenseRequest) =>
    apiClient<ExpenseResponse>('/expenses', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  cancel: (id: number) =>
    apiClient<ExpenseResponse>(`/expenses/${id}/cancel`, { method: 'POST' }),
}

File: BE/src/main/java/Manager_vnd/Manager/feature/expense/ExpenseService.java
package Manager_vnd.Manager.feature.expense;

import java.time.LocalDate;

import Manager_vnd.Manager.dto.PaginatedResult;
import Manager_vnd.Manager.feature.expense.dto.CreateExpenseRequest;
import Manager_vnd.Manager.feature.expense.dto.ExpenseResponse;

public interface ExpenseService {

    PaginatedResult<ExpenseResponse> getAll(
            int page, int size, String sort,
            Long categoryId, ExpenseStatus status,
            LocalDate fromDate, LocalDate toDate);

    ExpenseResponse getById(long id);

    ExpenseResponse create(CreateExpenseRequest request);

    ExpenseResponse cancel(long id);
}
